### DISCUSSION TITLE
Embedding tweets in docs

### DISCUSSION THREAD

**Discussion by aapnetin** [outline/outline#3528] [General] [answered]
Is it possible to embed tweets in docs ?
If yes, what happens when the author of the tweet deletes ?

Similarly what happens to embedded Google Docs and YouTube videos when their respective owners deletes them ?

Outline docs maintains local copy of embedded content ?

**Comment by Limezy** [comment]
Embedded content is basically an iFrame.
So no, Outline **doesn't maintain any local copy of the embedded content.** It will only store the URL of that content.
You may have some of the integrated services where basically the whole data is within the URL, but this is rare (and not the case for any of the above mentionned services)

**Reply by aapnetin** [reply]
Outline says : 
![image](https://user-images.githubusercontent.com/103986280/168467655-ba76fe29-2336-4366-9570-83c2d37b7ec4.png)

but nothing happens when i paste html into document ! 
![image](https://user-images.githubusercontent.com/103986280/168468034-0767bb17-f25d-4c16-b839-395bd9562223.png)

What i am doing wrong in pasting html ?

**Reply by tommoor** [reply]
Well, only simple HTML is supported - you can't just paste script tags in. It also depends on the paste source, as we have to detect it as html – otherwise it will be interpreted as markdown which might be what happened here.

The most reliable way to import html is to import a .html file rather than paste, but it would enable twitter embeds as iframes and script tags are stripped on import.

**Comment by Limezy** [comment, marked as answer]
And for the twitter part of your question, there was a PR to try implementing it, but sadly it was closed. https://github.com/outline/outline/pull/2491
